tools/linker.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import shutil
+
+
+def create_link(source, destination, fallback_copy=False):
```

Comment:
This seems like rather advanced logic just to create some symlinks. What edge case is it accounting for that I'm not seeing?

Reply:
Handling of different platforms/setups. This is related to my previous response to your comment on using simpler logic for symlinking.